cvr/src/main/java/org/cbioportal/cmo/pipelines/CVRPipeline.java

```diff
@@ -182,43 +191,72 @@ private static void checkExceptions(JobExecution jobExecution, JobParameters job
+                if (!option.getOpt().equals("t") && !option.getOpt().equals("c") && !option.getOpt().equals("g") && !option.getOpt().equals("z")) {
+                    String error_message = String.format("The --consume_samples option is only compatible with the --test and --gml and --drop_dead_instant options. You used an incompatible option (--%s/-$s)", option.getLongOpt(), option.getOpt());
                     log.error(error_message);
-                    help(options,1);
+                    help(options, 1);
                 }
             }
-            launchConsumeSamplesJob(args, commandLine.getOptionValue("c"), commandLine.hasOption("t"), commandLine.hasOption("g"));
-        }
-        else {
-            Integer maxNumSamplesToRemove = CVRUtilities.DEFAULT_MAX_NUM_SAMPLES_TO_REMOVE;
-            try {
-                if (commandLine.hasOption("r")) {
-                    maxNumSamplesToRemove = Integer.valueOf(commandLine.getOptionValue("r"));
+        } else {
+            // unless we are doing a consume job (which only requires a json filename) check for required arguments
+            if (!commandLine.hasOption("d") || !commandLine.hasOption("p") || !commandLine.hasOption("i")) {
+                help(options, 0);
+            }
+            if (commandLine.hasOption("r")) {
+                try {
+                    Integer.valueOf(commandLine.getOptionValue("r"));
+                } catch (NumberFormatException e) {
+                    e.printStackTrace();
+                    throw new RuntimeException("Cannot parse argument as integer: " + commandLine.getOptionValue("r"));
                 }
             }
-            catch (NumberFormatException e) {
-                e.printStackTrace();
-                throw new RuntimeException("Cannot parse argument as integer: " + commandLine.getOptionValue("r"));
+        }
+        if (commandLine.hasOption("z")) {
+            String dateString = commandLine.getOptionValue("z");
+            String dropDeadInstantString = InstantStringUtil.convertToIsoInstantFormat(dateString);
+            if (dropDeadInstantString == null) {
+                throw new RuntimeException("Cannot parse argument as valid date string for --drop_dead_instant: " + commandLine.getOptionValue("z"));
+            }
+            Instant instant = InstantStringUtil.createInstant(dropDeadInstantString);
+            if (instant == null) {
+                throw new RuntimeException("Instant could not be creatd from ISO_INSTANT formatted date time string : " + dropDeadInstantString);
+            }
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        Options options = CVRPipeline.getOptions(args);
+        CommandLineParser parser = new DefaultParser();
+        CommandLine commandLine = parser.parse(options, args);
+        validateArguments(options, commandLine);
+        String dropDeadInstantString = "";
```

Comment:
In this case `dropDeadInstantString` has a length of 0, and will be set to a distant time.